README.md

```diff
@@ -7,11 +7,139 @@ maintaining rather than effacing provenance.
 
 ## How to use it?
 
-This is very early-stage work. The file `jobs/WeCount-ODC.json5` contains a very simple two-element
+This is very early-stage work. The file `jobs/WeCount-ODC.json5` contains a very simple three-element
 data pipeline which will check out git repositories to use as inputs for a "forgiving data merge".
 Merged outputs together with provenance information linking back to the source data will be written
 into directory `dataOutput`.
 
 Run the sample pipeline by running
 
     node driver.js
+
+## What does it contain?
+
+This package contains two primary engines - firstly, the "tangled mat" structure which is used to track the provenance
+in overlaid data structures, and secondly the data pipeline which orchestrates tasks consuming and producing CSV files
+also tracking provenance.
+
+There is also a sample pipeline consuming data from [https://github.com/inclusive-design/covid-assessment-centres],
+performing an outer right join, and synthesizing accessibility data suitable for visualisation with
+[https://github.com/inclusive-design/covid-data-monitor].
+
+### The tangled mat
+
+The basic operation of the tangled mat is illustrated in diagram [Tangled Mat](https://docs.google.com/drawings/d/1OIT6zN0jwwuyt4ZmFeA-eWJiFEert8uM2zLtReqkve0/edit).
+Several layers of arbitrary JSON shape may be combined, with a resulting structure which is the result of a deep merge
+as if they had been combined using
+
+    jQuery/fluid.merge(true, []/{}, layer1, layer2 ...)
+
+With the difference that
+
+1. The merging is performed in a lazy manner - a result in the merged structure is only evaluated at the point it is
+demanded,
+2. The provenance of each final object in the merged structure is available in a "provenance map" which is isomorphic
+to the merged value, recording which layer each resulting leaf value was drawn from
+
+The primary API of the mat consists of its creator function `fluid.tangledMat`, and methods `addLayer`, `getRoot`,
+`readMember`, `getProvenance`, `evaluateFully` and `setWritableLayer` - see JSDocs. An example session interacting with the
+mat mirroring the example drawing:
+
+````javascript
+var mat = fluid.tangledMat([{
```

Comment:
What if different layers contain the same element, for example, key `a` exists in "Layer2" or/and "Layer3"? In this case, is the plan to ask for the user intervention or allow defining the precedence at adding layers? Is this use case not yet considered in this version?

Reply:
Values in higher layers always override those in lower ones, just as with $.extend(true ....)  - I've added a test case to clarify this